checker.cpp

```diff
@@ -1,17 +1,101 @@
-#include <assert.h>
-
-bool vitalsAreOk(float bpm, float spo2, float respRate) {
-  if(bpm < 70 || bpm > 150) {
-    return false;
-  } else if(spo2 < 90) {
-    return false;
-  } else if(respRate < 30 || respRate > 95) {
-    return false;
-  }
-  return true;
+#include<iostream>
+#include<assert.h>
+using namespace std;
+
+class AlertBasetype {
+public:
+	virtual void raise(const char* vitalname, const char* level) = 0;
+};
+
+class AlertSMS : public AlertBasetype {
+public:
+	void raise(const char* vitalname, const char* level) {
+		cout << "SMS: " << vitalname << " is " << level << " !" << endl;
+	}
+};
+
+class AlertSound : public AlertBasetype {
+public:
+	void raise(const char* vitalname, const char* level) {
+		cout << "Sound: " << vitalname << " is " << level << " !" << endl;
+	}
+};
+
+class VitalCheck
+{
+private:
+	const char* _name;
+	float _value;
+	
+	float _upper_limit;
+	float _lower_limit;
+	const char* _level;
+
+	AlertBasetype* _alert;
+
+public:
+	VitalCheck();
+	VitalCheck(const char*, float, float, float, AlertBasetype* alert);
+	
+	void SetLevel();
+	bool Level_is_ok();  
+};
+VitalCheck::VitalCheck() : _value(0), _upper_limit(0), _lower_limit(0) {}
+VitalCheck::VitalCheck(const char* name, float value, float lower_limit, float upper_limit, AlertBasetype* alert) :
+_name(name), _value(value), _lower_limit(lower_limit), _upper_limit(upper_limit), _alert(alert)
+{
+	SetLevel();
+	Level_is_ok();
+}
+void VitalCheck::SetLevel()
+{
+	if (_value < _lower_limit)
+		_level = "LOW";
+	else if (_value > _upper_limit)
+		_level = "HIGH";
+	else
+		_level = "OK";
 }
+bool VitalCheck::Level_is_ok()
+{
+	if (_level != "OK")
+	{
+		_alert->raise(_name, _level);
+		return false;
+	}
+	else
+		return true;	
+}
+
+class PatientVitals
+{
+private:
+	VitalCheck _bpm;
+	VitalCheck _spo2;
+	VitalCheck _respRate;
+public:
+	PatientVitals(float bpm, float spo2, float resprate, AlertBasetype* alert);
+};
+
+PatientVitals::PatientVitals(float bpm, float spo2, float respRate, AlertBasetype* alert)
+{
+	VitalCheck _bpm("BPM", bpm, 70, 150, alert);
```

Comment:
I don't think this will change the member variable. It declares another local variable called _bpm, which hides the member variable within this function. The _bpm member variable will be initialized with the default constructor.
Same with the other two variables.
Write a test in main() and check it before fixing.
Try an initialization list to initialize the data-members.